### PR TITLE
fix: supervision updates after ied rename

### LINKED ISSUE #1224: On rename in IED editor, supervision is not updated
**Describe the bug**

Supervision instances (`lnClass` = `LGOS` or `LSVS`) have the control block reference of the publishing device included in the `Val` of the `setSrcRef` `DAI`. For example:

```xml
<DOI name="GoCBRef">
    <DAI name="setSrcRef">
        <Val>XAT_232_M1MUGO/LLN0.Ind</Val>
    </DAI>
</DOI>
```

When we rename the device in the IED editor, this is not updated.

**To Reproduce**

1. Open a file with supervisions.
2. Rename the IED.
3. Check to see if the supervision information is updated.
4. Find that it is not.

**Expected behavior**

When an IED is renamed, the subscribing supervision instances are updated.

If I rename a device from: `XAT_BusA_P2` to `BlueSky`, then I expect the following change in all LGOS or LSVS LNs (to `GoCBRef` or `SvCBRef`)

```xml
<DOI name="GoCBRef" desc="Ref. GOOSE CB">
    <DAI name="setSrcRef" valImport="true">
        <Val>XAT_BusA_P2Bay240_DCB/LLN0.Ctl_1</Val>
    </DAI>
</DOI>
```

to:

```xml
<DOI name="GoCBRef" desc="Ref. GOOSE CB">
    <DAI name="setSrcRef" valImport="true">
        <Val>BlueSkyBay240_DCB/LLN0.Ctl_1</Val>
    </DAI>
</DOI>
```

**Additional context**

We do the renaming in `src/wizards/foundation/actions.ts` in `updateNamingAttributeWithReferencesAction` and call it from `src/wizards/ied.ts` (about L#212 currently).

### PR DESCRIPTION
**Pull request by Stef3st**
closes #1224

### CODE REVIEW

File: packages/open-scd/test/unit/wizards/foundation/references.test.ts [hunk 1]
```diff
@@ -103,6 +112,44 @@ describe('Update reference for ', () => {
...
+        '.' +
+        input3.getAttribute('srcCBName');
+
+      const oldValName3 = oldName + valSuffix3;
+
+      expectUpdateTextValue(
+        updateActions[4],
+        'DAI',
+        oldValName3,
+        newName + valSuffix3
```
Comment (JakobVogelsang) [inline packages/open-scd/test/unit/wizards/foundation/references.test.ts:149]: Why not using the `newValueName` that is defined in the beginning of the test?

**Review by JakobVogelsang — CHANGES_REQUESTED** [review]
I would have love to have the changes be separated somehow from the prettier changes. Them be in the same commit makes the review quite complicated.

**Comment by JakobVogelsang** [comment]
I have not checked out yet and did not do the functional review.

File: packages/open-scd/test/unit/wizards/foundation/references.test.ts (hunk 1 above)
Comment (Stef3st) [reply packages/open-scd/test/unit/wizards/foundation/references.test.ts:149]: oldValName and newValName were to static. Its something I ended up not using anymore but forgot to remove. Good catch!

File: packages/open-scd/src/wizards/foundation/references.ts [hunk 2]
```diff
@@ -190,13 +240,86 @@ export function updateReferences(element: Element, oldName: string | null, newNa
...
+        if (oldName + suffixCBReference === value.textContent!.trim()) {
+          const newElement = cloneElementAndTextContent(
+            value,
+            newName + suffixCBReference
+          );
+          actions.push({
+            old: { element: value },
+            new: { element: newElement },
+          });
+          break extRefLoop;
```
[the remark this answers is not in the document]
Comment (Stef3st) [reply packages/open-scd/src/wizards/foundation/references.ts:316]: I agree, I got in my head with too many corner case that I couldn't figure out how to deal with them all in 2 loops. Ill try to think of a better way if this is unacceptable

File: packages/open-scd/src/wizards/foundation/references.ts [hunk 3]
```diff
@@ -190,13 +240,86 @@ export function updateReferences(element: Element, oldName: string | null, newNa
...
+) {
+  // Each IED will be checked for the extRef elements. So firstly all the IEDs will be gathered with the querySelectorAll function.
+  const ieds = element.ownerDocument.querySelectorAll('IED');
+  ieds.forEach(ied => {
+    // All Val elements inside LGOS and LSVS lnClasses that starts with the IED name that needs to be changed will be gathered.
+    // Because of a very rare case where multiple IED start with the same name, all will be gathered.
+    // If none are found continue to the next IED.
+    const valValues: Element[] = Array.from(
+      ied.querySelectorAll(
+        `LN[lnClass="LGOS"] > DOI > DAI > Val, LN[lnClass="LSVS"] > DOI > DAI > Val[textContent*="${oldName}"]`
```
[the remark this answers is not in the document]
Comment (Stef3st) [reply packages/open-scd/src/wizards/foundation/references.ts:277]: Will do!

File: packages/open-scd/src/wizards/foundation/references.ts [hunk 4]
```diff
@@ -190,13 +240,86 @@ export function updateReferences(element: Element, oldName: string | null, newNa
...
           const newElement = cloneElementAndTextContent(element, newName);
-          actions.push({old: {element}, new: {element: newElement}});
-        })
+          actions.push({ old: { element }, new: { element: newElement } });
+        });
     }
-  })
+  });
+
+  if (element.tagName === 'IED') updateVals(element, oldName, newName, actions);
```
[the remark this answers is not in the document]
Comment (Stef3st) [reply packages/open-scd/src/wizards/foundation/references.ts:248]: Will do that!

File: packages/open-scd/src/wizards/foundation/references.ts [hunk 5]
```diff
@@ -190,13 +240,86 @@ export function updateReferences(element: Element, oldName: string | null, newNa
...
+ * @param newName - The new name of the element.
+ * @param actions - Array of Replace actions from the updateReferences function. The Val elements will be added to this array.
+ */
+function updateVals(
+  element: Element,
+  oldName: string | null,
+  newName: string,
+  actions: Replace[]
+) {
+  // Each IED will be checked for the extRef elements. So firstly all the IEDs will be gathered with the querySelectorAll function.
```
[the remark this answers is not in the document]
Comment (Stef3st) [reply packages/open-scd/src/wizards/foundation/references.ts:269]: No its not, I just wanted to make this as clear as possible.

File: packages/open-scd/src/wizards/foundation/references.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Stef3st) [reply packages/open-scd/src/wizards/foundation/references.ts:277]: 1. Good catch
2. Im not sure what you mean? I check if the textContent contains the oldName

**Review by JakobVogelsang — APPROVED** [review]
LGTM

**Pull request merged**